aiohttp/web_protocol.py

```diff
@@ -453,7 +453,10 @@ async def _handle_request(
             finally:
                 self._current_request = None
         except HTTPException as exc:
-            resp = exc
+            resp = Response(
+                status=exc.status, reason=exc.reason, text=exc.text, headers=exc.headers
+            )
+            resp._cookies = exc._cookies
```

Comment:
I have a strong feeling this code is rather fiddly and probably shouldn't be touched..